### ISSUE TITLE
Shims dont work if same executable is provided by many plugins

### ISSUE THREAD

**Issue by vic** [asdf-vm/asdf#431]
### Steps to reproduce

Install two tools that create shims for the same executable. For example both `nodejs` and `graalvm` provide their respective `npm` executable. Same should happen with `java` and `graalvm`.

In project A:

```shell
cd A
echo graalvm 1.0.0-rc5 > .tool-versions
asdf install
```

In project B:

```shell
cd B
echo nodejs 10.7.0 > .tool-versions
asdf install
```

### Expected behavior

In project A, executing `npm` should use the `graalvm` executable.
In project B, executing `npm` should use the `nodejs` executable.


### Actual behavior

In project B, executing `npm` fails with asdf complaining that no version is set for `graalvm`.

This is because the `npm` shim was created first for the `graalvm` plugin. A horrible workaround was to delete my `~/.asdf/shims/{npm,node,js}` and `asdf reshim nodejs`. But this would be needed everytime I switch between project A and B.

### Proposed solution

Shims should determine which tool I'm trying to invoke.

Currently the shims we generate have the following lines:

`cat ~/.asdf/shims/npm`                                                                                                                                                                                                              
```shell
#!/usr/bin/env bash
# asdf-plugin: graalvm
# asdf-plugin-version: 1.0.0-rc7
# asdf-plugin-version: 1.0.0-rc5
exec ~/.asdf/bin/private/asdf-exec graalvm bin/npm "$@"
```

Note that `asdf-exec` takes `graalvm` as the first argument. This prevents the same shim to work with other tools like `nodejs`.

I'm proposing we change the meta comments `asdf-plugin-version` to something like:

```shell
#!/usr/bin/env bash
# asdf-plugin: graalvm 1.0.0-rc7
# asdf-plugin: graalvm 1.0.0-rc5
# asdf-plugin: nodejs 10.7.0
exec ~/.asdf/bin/private/asdf-exec npm "$@"
```

And `asdf-exec` no longer taking a tool-name as first-arg.

When the current project (like `B`) specifies a tool version (like nodejs-10.7.0, but no graalvm), then `asdf-exec` should find out that it needs to execute nodejs's npm binary.


### Environment

**OS**: OSX

**asdf version**: master

**Comment by vic** [comment]
Should be solved by #432

**Comment by Stratus3D** [comment]
I like the solution you have proposed here. Great work @vic!
